src/shared/fix-query-parser/query.ts

```diff
@@ -959,6 +1058,15 @@ export class Query {
     })
   }
 
+  public delete_all_term_if_any(): Query {
+    const existing = this.working_part.term.find_terms((i) => i instanceof AllTerm)
+    return existing.length > 0
+      ? produce(this, (draft) => {
+          draft.working_part.term = draft.working_part.term.delete_terms((t) => t instanceof AllTerm)
+        })
+      : this
+  }
+
```

Comment:
can be removed

Reply:
No This is to remove AllTerm when there's one. this solution is for when we're trying to add a term to a query when there's a all in it